sensing/autoware_downsample_filters/src/filter.cpp

```diff
@@ -0,0 +1,344 @@
+    pub_output_ = this->create_publisher<PointCloud2>(
+      "output", rclcpp::SensorDataQoS().keep_last(max_queue_size_), pub_options);
+  }
+
+  // TODO(sykwer): Change the corresponding node to subscribe to `faster_input_callback`
+  // each time a child class supports the faster version.
+  // When all the child classes support the faster version, this workaround is deleted.
+  std::set<std::string> supported_nodes = {"VoxelGridDownsampleFilter"};
+  auto callback = supported_nodes.find(filter_name) != supported_nodes.end()
+                    ? &Filter::faster_input_callback
+                    : &Filter::input_callback;
+
+  // Subscribe in an old fashion to input only (no filters)
+  // CAN'T use auto-type here.
+  std::function<void(const PointCloud2ConstPtr msg)> cb =
+    std::bind(callback, this, std::placeholders::_1);
+  sub_input_ = create_subscription<PointCloud2>(
+    "input", rclcpp::SensorDataQoS().keep_last(max_queue_size_), cb);
+
+  // Set tf_listener, tf_buffer.
+  setupTF();
+
+  published_time_publisher_ = std::make_unique<autoware_utils_debug::PublishedTimePublisher>(this);
+  RCLCPP_DEBUG(this->get_logger(), "[Filter Constructor] successfully created.");
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+void autoware::downsample_filters::Filter::setupTF()
+{
+  transform_listener_ = std::make_unique<autoware_utils_tf::TransformListener>(this);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// TODO(sykwer): Temporary Implementation: Delete this function definition when all the filter nodes
+// conform to new API.
+void autoware::downsample_filters::Filter::compute_publish(const PointCloud2ConstPtr & input)
+{
+  auto output = std::make_unique<PointCloud2>();
+
+  // Call the virtual method in the child
+  filter(input, *output);
+
+  if (!convert_output_costly(output)) return;
+
+  // Copy timestamp to keep it
+  output->header.stamp = input->header.stamp;
+
+  // Publish a boost shared ptr
+  pub_output_->publish(std::move(output));
+  published_time_publisher_->publish_if_subscribed(pub_output_, input->header.stamp);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+// TODO(sykwer): Temporary Implementation: Delete this function definition when all the filter nodes
+// conform to new API.
+void autoware::downsample_filters::Filter::input_callback(const PointCloud2ConstPtr cloud)
+{
+  // If cloud is given, check if it's valid
+  if (!isValid(cloud)) {
+    RCLCPP_ERROR(this->get_logger(), "[input_callback] Invalid input!");
+    return;
+  }
+
+  /// DEBUG
+  RCLCPP_DEBUG(
+    this->get_logger(),
+    "[input_callback] PointCloud with %d data points and frame %s on input topic "
+    "received.",
+    cloud->width * cloud->height, cloud->header.frame_id.c_str());
+  ///
```

Comment:
It's obvious that it's a debug info so we don't need the comment
```suggestion
  RCLCPP_DEBUG(
    this->get_logger(),
    "[input_callback] PointCloud with %d data points and frame %s on input topic "
    "received.",
    cloud->width * cloud->height, cloud->header.frame_id.c_str());
```